brave/brave-core
# feat(wallet): Cancel & Speed Up Tx in Tx Status Screen

Resolves https://github.com/brave/brave-browser/issues/41799
Resolves https://github.com/brave/brave-browser/issues/41800



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Create either a `.ethSend`, `.erc20Approval`, `.erc20Transfer`, `.eth721Transfer` or `.ethSwap`
2. Once the tx is submitted to the chain, you are in the flow of displaying the current tx's status
3. You will see `Cancel transaction` button enabled once the tx has been submitted 
4. Quickly click this button you will see a confirmation screen shows up 
5. Click the back button the cancel 
6. You might need to repeat step 1 to 4 if your transaction gets confirmed 
7. Click `Cancel transaction` in the confirmation 
8. The status screen will be dismissed and a new pending transaction created 
9. Click confirm
10. Now you are entering again the tx status flow but for this new transaction 
11. Notice there is no `Cancel transaction` button anymore 
12. Let it gets confirmed 
13. You will see the confirmed status will display correct message.
14. Go back to `Activity` tab 
15. You will see a transaction that sent 0 ETH is successfully confirmed on chain
16. Create other transaction that is not Eth coin type
17. You will not see the `Cancel transaction` button in the tx status flow

Cancel button | Confirmation | Activity
--- | --- | ---
![Simulator Screenshot - iPhone 16 Pro Max - 2024-12-06 at 14 35 08](https://github.com/user-attachments/assets/1352220a-dba0-451d-b8de-31c24eacf292) | ![Simulator Screenshot - iPhone 16 Pro Max - 2024-12-09 at 17 52 37](https://github.com/user-attachments/assets/c31f0554-2070-40e0-a837-95eab991b395) | ![Simulator Screenshot - iPhone 16 Pro - 2024-12-04 at 14 03 30](https://github.com/user-attachments/assets/79c88e92-6131-4a57-94c6-0ac104bf17bb)
Cancellation Tx | Send Tx Cancelled | ERC20Approval Cancelled
![Simulator Screenshot - iPhone 16 Pro Max - 2024-12-06 at 14 35 08](https://github.com/user-attachments/assets/6b358182-3e93-4365-8aea-2cf7041eb2e8) | ![Simulator Screenshot - iPhone 16 Pro Max - 2024-12-06 at 14 35 22](https://github.com/user-attachments/assets/ed3297a6-857d-47e2-83f7-11263e80deb3) | ![Simulator Screenshot - iPhone 16 Pro Max - 2024-12-06 at 15 06 19](https://github.com/user-attachments/assets/3dc011ab-17b6-4c1a-85a1-d700e95df658)
ERC20 Tx Cancelled | |
![Simulator Screenshot - iPhone 16 Pro Max - 2024-12-06 at 15 43 31](https://github.com/user-attachments/assets/16655c14-8a8d-41b1-9a16-e9faa359ee56) | |

Speed up a eth tx
For demo purpose the banner will be displayed 1 second after tx status screen is visible. But the actual timer is 5 seconds. If you tx is confirmed or errored out before 5 seconds you won't see the speed up banner
https://github.com/user-attachments/assets/fd653976-db8b-4771-b55b-5291ad710fcd
1. create a ETH tx and confirm it. Please take a glance of the gas fee.
2. once the tx status shows up, wait for the speed up banner to be shown 
3. click the speed up button
4. you will see the tx status screen will be dismissed and a new pending tx will be shown and notice that this pending tx's gas fee should be greater than the gas fee in step 1
5. confirm this pending tx
6. tx should get confirmed shortly.
